habitat_baselines/evaluate_ppo.py

```diff
@@ -149,13 +200,153 @@ def main():
         episode_counts += 1 - not_done_masks
         current_episode_reward *= not_done_masks
 
+        next_episodes = envs.current_episodes()
+        envs_to_pause = []
+        n_envs = envs.num_envs
+        for i in range(n_envs):
+            if next_episodes[i].episode_id in stats_episodes:
+                envs_to_pause.append(i)
+
+            # episode ended
+            if not_done_masks[i].item() == 0:
+                stats_episodes.add(current_episodes[i].episode_id)
+                generate_video(
+                    args,
+                    rgb_frames[i],
+                    current_episodes[i].episode_id,
+                    checkpoint_idx,
+                    infos[i]["spl"],
+                    writer,
+                )
+                rgb_frames[i] = []
+
+            # episode continues
+            elif args.video_option != "no_video":
```

Comment:
This should just be an `if`.  ATM the video won't contain the first frame.

Reply:
I can see how this can cause a problem. But with the current implementation, instead of missing the first frame, should it be missing the last frame? Making it an `if` block and move it before the `if` block for "episode ended" should fix the issue I think.

Reply:
When the stop action is called, the environment returns the observations for the start of the next episode (assuming auto reset is turned on). The last frame is the frame that lead to stop being taken, and that’s already being recorded.

Reply:
@erikwijmans I see. The issue now is that when stop action is called, observation and info returned seem to be from different episodes (observation is from next episode, info from cur episode). This causes mis-match of topdown map and egocentric views. A video demo sent privately.

Reply:
Reverted back to skipping first frame, as a tmp fix. Will need further discussion (and potential changes to how top-down-map is represented) for a full fix.